### PR TITLE
Add metadata comparison tests

### PR DESCRIPTION
**Pull request by PavelPunegov**

### CODE REVIEW

File: backend.native/tests/build.gradle [hunk 1]
```diff
@@ -240,6 +240,7 @@ run {
     dependsOn(tasksOf(KonanTest))
     // Add framework tests
     dependsOn(tasksOf(FrameworkTest))
+    dependsOn(tasksOf(MetadataComparisonTest))
```
Comment (SvyatoslavScherbina) [inline backend.native/tests/build.gradle:243]: We might need to add this test to sanity.
Also there are probably other test kinds that don't run with `run` or `sanity`. For example, `CoverageTest`.

Comment (PavelPunegov) [reply backend.native/tests/build.gradle:243]: I'm looking through all our tests to find out what else isn't run in a `run {}` and in `sanity {}` but could have been.

**Comment by SvyatoslavScherbina** [comment]
@PavelPunegov Do you plan to [include](https://github.com/JetBrains/kotlin-native/pull/4222#discussion_r435343037) coverage (and maybe other) tests with a separate PR?

**Comment by PavelPunegov** [comment]
> @PavelPunegov Do you plan to [include](https://github.com/JetBrains/kotlin-native/pull/4222#discussion_r435343037) coverage (and maybe other) tests with a separate PR?

No, I've just find out that you have merged your PR. I am going to add it here then.

**Pull request merged**